### ISSUE TITLE
Named imports from CJS modules with * re-exports are failing

### ISSUE THREAD

**Issue by teug91** [dhis2/app-platform#69]
```js
import React from 'react'
import { Table } from '@dhis2/ui-core'

export const App = () => <Table />
```

This gives the following error:
```
./src/current-d2-app/app.js
Attempted import error: 'Table' is not exported from '@dhis2/ui-core' (imported as 'Table$1').
```

```js
import { BrowserRouter } from 'react-router-dom'
```

This gives the following error, just from importing:
```
Error: 'BrowserRouter' is not exported by node_modules/react-router-dom/index.js
```

This works:
```js
import { Table } from '@dhis2/ui-core/build/cjs/Table'
```

Both the `Table` and `BrowserRouter` example are working as intended in apps initialized with CRA. I have not come across other examples of this issue.

```
@dhis2/cli-app-scripts@1.4.3
@dhis2/ui-core@3.11.0
react-router-dom@5.1.0

react-scripts@3.1.2
```

**Comment by amcgee** [comment]
Thanks for the report @teug91!

I believe this has to do with CJS vs ES bundles - we unfortunately only support static named exports from CJS bundles right now - CRA can resolve dynamic named exports because webpack does module resolution at runtime, rather than statically at compile time.  See [this comment](https://github.com/dhis2/app-platform/blob/master/cli/config/rollup.config.js#L77) and the [underlying rollup issue](https://github.com/rollup/rollup-plugin-commonjs/issues/211#issuecomment-337897124)

This is the reason the Table import fails (it's exported as `export * from ./Table.js`) while other `ui-core` components should work fine.

I'll look into this a bit more, I'd really like to solve this problem

**Comment by amcgee** [comment]
@teug91 just to follow up here, another method that should work in the Table case is to use the default export from the CJS bundle, like so -

```js
import uiCore from '@dhis2/ui-core'
const { Table } = uiCore
```

**Comment by amcgee** [comment]
@teug91 another update - the original code you tried to use (`import { Table } from '@dhis2/ui-core'`) should work now, since we removed the dynamic exports from `ui-core` (https://github.com/dhis2/ui-core/pull/459) and upgraded `ui-core` in platform version [1.5.0](https://github.com/dhis2/app-platform/releases/tag/v1.5.0)

**Comment by amcgee** [comment]
@varl unfortunately this is still something we need to fix, it doesn’t only affect ui-core

**Comment by varl** [comment]
Ah, alright. :+1:

**Comment by teug91** [comment]
Just to confirm, this workaround does indeed work:
```js
import routing from 'react-router-dom'
const { HashRouter, Route } = routing
```

**Comment by amcgee** [comment]
Fixed in #187
